lib/resource/resource.ex

```diff
@@ -2164,10 +2170,16 @@ defmodule AshGraphql.Resource do
   end
 
   defp resource_filter_fields(resource, schema) do
-    boolean_filter_fields(resource, schema) ++
-      attribute_filter_fields(resource, schema) ++
-      relationship_filter_fields(resource, schema) ++
-      aggregate_filter_fields(resource, schema) ++ calculation_filter_fields(resource, schema)
+    all_fields =
+      boolean_filter_fields(resource, schema) ++
+        attribute_filter_fields(resource, schema) ++
+        relationship_filter_fields(resource, schema) ++
+        aggregate_filter_fields(resource, schema) ++ calculation_filter_fields(resource, schema)
+
+    filterable_fields = AshGraphql.Resource.Info.filterable_fields(resource)
+
+    all_fields
+    |> Enum.filter(&(is_nil(filterable_fields) or &1.identifier in filterable_fields))
```

Comment:
unfortunately, we can't actually do this using the identifier, because it may have been remapped using the `field_names` option. So we'll want to do it in the individual functions.